### ISSUE TITLE
Live stream not showing or encoding

### ISSUE THREAD

**Issue by guymass** [WWBN/AVideo#8829]
I am trying to run my live stream on my site over a reverse proxy connection I have setup. I have the site loading and also the socket is connecting. I can connect and start a live stream with my OBS, however notification doesn't show any lie stream on the site and also once I close my stream I do not see any encoding process and the file is just lost to the void somewhere idk.

Here are few screen shots from the server and the encoder, maybe you can help me to figure out what is the problems.
Also I have all ports open and tunneled from my proxy server to the local tunnel connection, maybe I need to change them to something else? I am using HAProxy as well to solve this issue but with no success. What strange is that I did use to have live stream working over my reverse proxy connection before, but this is a fresh install of a former damaged system I just finish recovering and I am trying to also reconfigure my live streams to work as well.

![Screenshot 5784-05-29 at 18 04 09](https://github.com/WWBN/AVideo/assets/58882903/1acdac3a-f0d8-4149-bc88-9bc9f7536d84)
![Screenshot 5784-05-29 at 18 03 58](https://github.com/WWBN/AVideo/assets/58882903/30c91bcd-de03-482d-b266-a75caab82896)
![Screenshot 5784-05-29 at 17 56 00](https://github.com/WWBN/AVideo/assets/58882903/6074ced9-a448-449b-a019-325422573ed9)
![Screenshot 5784-05-29 at 17 55 45](https://github.com/WWBN/AVideo/assets/58882903/f872e903-589a-4f37-9518-a091596ac7ed)
![Screenshot 5784-05-29 at 17 17 12](https://github.com/WWBN/AVideo/assets/58882903/64c02b6e-9ac6-4d5c-99be-304aec0ef797)

Thanks for any advice.

**Comment by guymass** [comment]
Also this is my current plugin configuration but I tested all sorts of ports and different addresses, I can't get the player to work or the live stream to be found by the RTMP server, I think :-).

![Screenshot 5784-05-29 at 18 12 23](https://github.com/WWBN/AVideo/assets/58882903/b207db21-e508-45f5-bd19-8a9f54fbd5ae)

**Comment by DanielnetoDotCom** [comment]
Definitely, this is something in your proxy

I am not sure how to resolve this, but this is the issue.

![image](https://github.com/WWBN/AVideo/assets/18009293/207da9d5-5dea-4bc9-934a-5aa9aa0b2b97)

**Comment by guymass** [comment]
I found that when I start my live stream the nginx server will try to connect to a wrong rtmp server of mine rtmp://fetube.flatearth.co.il/ instead of rtmp://feisafe.com/, I have search both in folders and databses and could find any reference to this url on this new installed/recovered site. Why is it connecting to a wrong server url? The stream key for my OBS is taken from the feisafe.com site and not from FETUBE site.

![Screenshot 5784-06-02 at 2 27 26](https://github.com/WWBN/AVideo/assets/58882903/e5e1cd30-4539-4007-8477-776a63e19aa9)

**Comment by guymass** [comment]
never mind sorry I had wrong stream url copied into my OBS

**Comment by guymass** [comment]
I am opening this again as that didn't solve my issue. I recompiled nginx again with openssl-1.1.1g library and now I am getting some strange errors in log showing me some strange strings which I think are encoded wrongly maybe. Here is the screenshot:

![Screenshot 5784-06-02 at 10 33 32](https://github.com/WWBN/AVideo/assets/58882903/6bc02c90-c297-470f-a703-8f0b1be68903)

**Comment by guymass** [comment]
I can now stream and see my stream through the shared URL, however it will not show on the site alerts or on front page.

![Screenshot 5784-06-02 at 10 51 36](https://github.com/WWBN/AVideo/assets/58882903/77818a7b-bcad-4145-8d8c-e88c46bf7404)

**Comment by guymass** [comment]
Now my live stream is stuck on the nginx server and even when I restart/stop/start the nginx server the live stream still appears and I cannot start a new live stream from OBS. I tried to click "Mark all as finished" but it doesn't work. How do I stop this live stream?

Showing even after nginx restart:

![Screenshot 5784-06-02 at 11 09 20](https://github.com/WWBN/AVideo/assets/58882903/f863d6f5-103d-4f2f-a82e-10dfd8e92e8c)

**Comment by guymass** [comment]
No idea why my live stream is not being encoded when live is finished.

2024/02/11 18:54:51 [info] 290729#0: *4 notify: play redirect to 'https://feisafe.com/AVideo/plugin/Live/on_play.php', client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:41 [info] 290729#0: *2 exec: child 290732 exited; ignoring, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:41 [info] 290729#0: *2 exec: terminating child 290732, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:41 [notice] 290729#0: signal 17 (SIGCHLD) received from 290732
2024/02/11 18:55:41 [notice] 290729#0: unknown process 290732 exited with code 1
2024/02/11 18:55:41 [info] 290729#0: epoll_wait() failed (4: Interrupted system call)
2024/02/11 18:55:41 [info] 290729#0: *4 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:41 [info] 290729#0: *4 disconnect, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:41 [info] 290729#0: *4 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:46 [info] 290729#0: *2 exec: starting managed child 'ffmpeg', client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:46 [info] 290729#0: *7 client connected '127.0.0.1'
2024/02/11 18:55:46 [info] 290729#0: *7 connect: app='live' args='' flashver='LNX 9,0,124,2' swf_url='' tc_url='rtmp://feisafe.com:1935/live' page_url='' acodecs=4071 vcodecs=252 object_encoding=0, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:46 [info] 290729#0: *7 createStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:46 [info] 290729#0: *7 play: name='https://feisafe.com/AVideo/plugin/Live/on_publish.php' args='' start=-2000 duration=0 reset=0 silent=0, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:46 [info] 290729#0: *7 notify: play 'feisafe.com/AVideo/plugin/Live/on_play.php', client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:55:46 [info] 290729#0: *7 notify: play redirect to 'https://feisafe.com/AVideo/plugin/Live/on_play.php', client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 notify: publish_done 'feisafe.com/AVideo/plugin/Live/on_publish_done.php', client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 exec: terminating child 290737, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 disconnect, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [notice] 290729#0: signal 17 (SIGCHLD) received from 290737
2024/02/11 18:56:18 [alert] 290729#0: unknown process 290737 exited on signal 9
2024/02/11 18:56:18 [info] 290729#0: epoll_wait() failed (4: Interrupted system call)
2024/02/11 18:56:18 [info] 290729#0: *7 disconnect, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *7 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
^C
root@feisafesrv:~/build# nano /etc/nginx/nginx.conf
root@feisafesrv:~/build# tail -f /etc/nginx/logs/*
2024/02/11 18:56:18 [info] 290729#0: *2 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 notify: publish_done 'feisafe.com/AVideo/plugin/Live/on_publish_done.php', client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 exec: terminating child 290737, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 disconnect, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *2 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [notice] 290729#0: signal 17 (SIGCHLD) received from 290737
2024/02/11 18:56:18 [alert] 290729#0: unknown process 290737 exited on signal 9
2024/02/11 18:56:18 [info] 290729#0: epoll_wait() failed (4: Interrupted system call)
2024/02/11 18:56:18 [info] 290729#0: *7 disconnect, client: 127.0.0.1, server: 0.0.0.0:1935
2024/02/11 18:56:18 [info] 290729#0: *7 deleteStream, client: 127.0.0.1, server: 0.0.0.0:1935

**Comment by guymass** [comment]
Ok I figured out my ffmpeg was not compiled properly with some missing libraries. I have recompiled it and now I can more information in my log. The following screenshot show another error with ffmpeg I am still not able to encode my live stream.

 ffmpeg   -user_agent "AVideoEncoder live.ahava528.com FFMPEG"  -i "https://storage1.ahava528.com/videos/_YPTuniqid_6014295f440cc5.20679795/_YPTuniqid_6014295f440cc5.20679795_HD.mp4" -map 0:v:0 -vf fps=1/1.81,scale=150:84:force_original_aspect_ratio=decrease,pad=150:84:-1:-1:color=black -frames:v 100  "/var/www/html/AVideo/Encoder/videos/thumbs_abcd3ebcb596a95e9c72dc3edf27c62a/out%03d.png"  2>&1 
[11-Feb-2024 22:32:35 Asia/Jerusalem] CreateSpirits: ERROR on dirsize=81920 ffmpeg   -user_agent "AVideoEncoder live.ahava528.com FFMPEG"  -i "https://storage1.ahava528.com/videos/_YPTuniqid_6014295f440cc5.20679795/_YPTuniqid_6014295f440cc5.20679795_HD.mp4" -map 0:v:0 -vf fps=1/1.81,scale=150:84:force_original_aspect_ratio=decrease,pad=150:84:-1:-1:color=black -frames:v 100  "/var/www/html/AVideo/Encoder/videos/thumbs_abcd3ebcb596a95e9c72dc3edf27c62a/out%03d.png"  2>&1

**Comment by guymass** [comment]
All my save file are still under /var/www/tmp and are not being published or encoded.

![Screenshot 5784-06-02 at 22 59 10](https://github.com/WWBN/AVideo/assets/58882903/d6c2f7ee-727f-49fa-b1dc-a951e3fbc9f7)

**Comment by guymass** [comment]
[11-Feb-2024 23:11:23 Asia/Jerusalem] AVideoLog::DEBUG: getVideosURL_V2(video_230528072600_v254b) Recreate cache requested [{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6597,"function":"getVideosURL_V2"},{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6582,"function":"isMediaFileMissing","class":"Video","type":"::"},{"file":"\/var\/www\/html\/AVideo\/objects\/videos.json.php","line":86,"function":"checkIfIsBroken","class":"Video","type":"::"}] SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:23 Asia/Jerusalem] AVideoLog::DEBUG: getVideosURL_V2(video_230529175117_v25ba) Recreate cache requested [{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6597,"function":"getVideosURL_V2"},{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6582,"function":"isMediaFileMissing","class":"Video","type":"::"},{"file":"\/var\/www\/html\/AVideo\/objects\/videos.json.php","line":86,"function":"checkIfIsBroken","class":"Video","type":"::"}] SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:23 Asia/Jerusalem] AVideoLog::DEBUG: getVideosURL_V2(video_230529185441_vba50) Recreate cache requested [{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6597,"function":"getVideosURL_V2"},{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6582,"function":"isMediaFileMissing","class":"Video","type":"::"},{"file":"\/var\/www\/html\/AVideo\/objects\/videos.json.php","line":86,"function":"checkIfIsBroken","class":"Video","type":"::"}] SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:24 Asia/Jerusalem] AVideoLog::DEBUG: Video::duration_in_seconds: id=82 00:00:00 video SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:24 Asia/Jerusalem] AVideoLog::DEBUG: Video::duration_in_seconds: id=83 00:00:00 video SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:24 Asia/Jerusalem] AVideoLog::DEBUG: Video::duration_in_seconds: id=84 00:00:00 video SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:24 Asia/Jerusalem] AVideoLog::DEBUG: getVideosURL_V2(video_230528072600_v254b) Recreate cache requested [{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6597,"function":"getVideosURL_V2"},{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6582,"function":"isMediaFileMissing","class":"Video","type":"::"},{"file":"\/var\/www\/html\/AVideo\/objects\/videos.json.php","line":86,"function":"checkIfIsBroken","class":"Video","type":"::"}] SCRIPT_NAME: /objects/videos.json.php
[11-Feb-2024 23:11:24 Asia/Jerusalem] AVideoLog::DEBUG: getVideosURL_V2(video_230529175117_v25ba) Recreate cache requested [{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6597,"function":"getVideosURL_V2"},{"file":"\/var\/www\/html\/AVideo\/objects\/video.php","line":6582,"function":"isMediaFileMissing","class":"Video","type":"::"},{"file":"\/var\/www\/html\/AVideo\/objects\/videos.json.php","line":86,"function":"checkIfIsBroken","class":"Video","type":"::"}] SCRIPT_NAME: /objects/videos.json.php

**Comment by DanielnetoDotCom** [comment]
Upon reviewing your current setup, it appears that you are utilizing a non-standard installation which unfortunately falls outside the scope of the support I can directly provide.

For instance, your system is configured to use PHP-FPM, whereas I generally recommend PHP as an Apache module for optimal compatibility and performance. 

Additionally, there seems to be a redirection mechanism in place within your Nginx or Apache configuration, likely from HTTP to HTTPS. This redirection needs to be removed to ensure proper functionality.

![image](https://github.com/WWBN/AVideo/assets/18009293/5a0c58c5-85e7-43c5-b27e-d3369b5a2ca9)

**Comment by guymass** [comment]
OK Well I have this site running off a reverse proxy and I did get it to work once before but I had to reinstall and move my server and also after updating it just doesn't work anymore. But I will take your suggestion and change the PHP-FPM although I always thought it was an essential part of the installation. I will close this for now unless I will get it to work again and then I will post my results for anyone who want to use the same kind of configuation.
